go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -217,11 +217,21 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 		j.Spec.MainClass = &sparkJob.MainClass
 	}
 
+	// Add Architecture Tolerations/NodeSelector to all pods
+	if taskTemplate.GetContainer().GetArchitecture() != core.Container_UNKNOWN {
+		j.Spec.Driver.NodeSelector = config.GetK8sPluginConfig().ArchitectureNodeSelector[strings.ToLower(taskTemplate.GetContainer().GetArchitecture().String())]
+		j.Spec.Driver.Tolerations = config.GetK8sPluginConfig().ArchitectureTolerations[strings.ToLower(taskTemplate.GetContainer().GetArchitecture().String())]
+		j.Spec.Executor.NodeSelector = config.GetK8sPluginConfig().ArchitectureNodeSelector[strings.ToLower(taskTemplate.GetContainer().GetArchitecture().String())]
+		j.Spec.Executor.Tolerations = config.GetK8sPluginConfig().ArchitectureTolerations[strings.ToLower(taskTemplate.GetContainer().GetArchitecture().String())]
+		j.Spec.SparkConf["spark.kubernetes.node.selector.beta.kubernetes.io/arch"] = "arm64"
```

Comment:
this is supported by core-spark afaik. The operator is supported to convert the Spec entries above to eventually spark conf like this but looks like there is a bug there.